microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/properties/PropertiesFileInlayHint.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.services.properties;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.eclipse.lsp4j.InlayHint;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+import org.eclipse.lsp4mp.commons.MicroProfileProjectInfo;
+import org.eclipse.lsp4mp.ls.commons.BadLocationException;
+import org.eclipse.lsp4mp.model.Node;
+import org.eclipse.lsp4mp.model.Node.NodeType;
+import org.eclipse.lsp4mp.model.PropertiesModel;
+import org.eclipse.lsp4mp.model.Property;
+import org.eclipse.lsp4mp.model.PropertyGraph;
+import org.eclipse.lsp4mp.model.PropertyValue;
+
+/**
+ * The properties file inlay hint support.
+ * 
+ * Given this properties file:
+ * 
+ * <code>
+    server.url=https://${host}:${port:8080}/${endpoint}
+	host=microprofile.io
+	app=project
+	service=eclipse/microprofile-config
+	endpoint=${app}/${service}
+ * </code>
+ * 
+ * Inlay hint will be displayed for the 2 properties which have expression:
+ * 
+ * <code>
+    server.url=https://${host}:${port:8080}/${endpoint} [https://microprofile.io:8080/project/eclipse/microprofile-config]
+	host=microprofile.io
+	app=project
+	service=eclipse/microprofile-config
+	endpoint=${app}/${service} [eclipse/microprofile-config]
+ * </code>
+ */
+
+class PropertiesFileInlayHint {
+
+	private static final Logger LOGGER = Logger.getLogger(PropertiesFileInlayHint.class.getName());
+
+	public List<InlayHint> getInlayHint(PropertiesModel document, MicroProfileProjectInfo projectInfo, Range range,
+			CancelChecker cancelChecker) {
+		List<InlayHint> hints = new ArrayList<>();
+		List<Node> children = document.getChildren();
+		for (Node child : children) {
+			cancelChecker.checkCanceled();
+			if (child.getNodeType() == NodeType.PROPERTY) {
+				Property property = (Property) child;
+				PropertyValue valueNode = property.getValue();
+				if (valueNode != null && valueNode.hasExpression()) {
+					// The current property has a value with expression:
+					// ex : server.url=https://${host}:${port:8080}/${endpoint}
+					PropertyGraph graph = new PropertyGraph(document);
```

Comment:
Do you want to do the resolving as is, or wait for #283?

Reply:
As you wish:) But perhaps it can be interested for #283 to have this inlay hint to test. Youcan see the resultof expand easily with inlay hint compare to hover. What do you think?

Reply:
I think merging before my PR makes sense if you want to demo it, since my PR might take a while to review. I will have to rewrite some of the code to resolve the expression in my PR.